lib/core/test/unit/Cardano/Wallet/Primitive/Types/UTxOSpec.hs

```diff
@@ -54,6 +77,61 @@ spec =
         it "filterByAddress is always subset" $
             property prop_filterByAddress_isSubset
 
+--------------------------------------------------------------------------------
+-- Filtering and partitioning
+--------------------------------------------------------------------------------
+
+prop_filter_disjoint :: (TxIn -> Bool) -> UTxO -> Property
+prop_filter_disjoint f u =
+    checkCoverage_filter_partition f u $
+    UTxO.filter f u `UTxO.disjoint` UTxO.filter (not . f) u === True
+
+prop_filter_partition :: (TxIn -> Bool) -> UTxO -> Property
+prop_filter_partition f u =
+    checkCoverage_filter_partition f u $
+    (UTxO.filter f u, UTxO.filter (not . f) u) === UTxO.partition f u
+
+prop_filter_toList :: (TxIn -> Bool) -> UTxO -> Property
+prop_filter_toList f u =
+    checkCoverage_filter_partition f u $
+    UTxO.toList (UTxO.filter f u) === L.filter (f . fst) (UTxO.toList u)
+
+prop_partition_disjoint :: (TxIn -> Bool) -> UTxO -> Property
+prop_partition_disjoint f u =
+    checkCoverage_filter_partition f u $
+    uncurry UTxO.disjoint (UTxO.partition f u) === True
+
+prop_partition_mappend :: (TxIn -> Bool) -> UTxO -> Property
+prop_partition_mappend f u =
+    checkCoverage_filter_partition f u $
+    uncurry (<>) (UTxO.partition f u) === u
+
+checkCoverage_filter_partition
+    :: Testable prop => (TxIn -> Bool) -> UTxO -> (prop -> Property)
+checkCoverage_filter_partition f u
+    = checkCoverage
+    . cover 10
+        (UTxO.filter f u `isNonEmptyProperSubsetOf` u)
+        "UTxO.filter f u `isNonEmptyProperSubsetOf` u"
+    . cover 10
+        (UTxO.filter (not . f) u `isNonEmptyProperSubsetOf` u)
+        "UTxO.filter (not . f) u `isNonEmptyProperSubsetOf` u"
+    . cover 10
+        (UTxO.size (UTxO.filter f u) > UTxO.size (UTxO.filter (not . f) u))
+        "UTxO.size (UTxO.filter f u) > UTxO.size (UTxO.filter (not . f) u)"
+    . cover 10
+        (UTxO.size (UTxO.filter f u) < UTxO.size (UTxO.filter (not . f) u))
+        "UTxO.size (UTxO.filter f u) < UTxO.size (UTxO.filter (not . f) u)"
+  where
+    u1 `isNonEmptyProperSubsetOf` u2 = True
```

Comment:
Is this True necessary?

Reply:
> Is this True necessary?

It's definitely not necessary.

It's just a stylistic choice so that we can place each condition on a line beginning with `&&`.

Reply:
@sevanspowell I've simplified these expressions in 19a0b7d669a2ecaa4f45a35941c08b27dfa48d07.